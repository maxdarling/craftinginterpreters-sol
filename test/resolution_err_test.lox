var a = "outer";
{
  var a = a;
}

fun bad() {
  var a = "first";
  var a = "second";

  var b = "unused";
  var c = "unused";
}

return "at top level";
