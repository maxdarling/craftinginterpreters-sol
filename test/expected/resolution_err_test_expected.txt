[line 3] Error at 'a': Can't read local variable in its own initializer.
[line 3] Error at 'a': Local variable is not used.
[line 8] Error at 'a': Already a variable with this name in this scope.
[line 8] Error at 'a': Local variable is not used.
[line 10] Error at 'b': Local variable is not used.
[line 11] Error at 'c': Local variable is not used.
[line 14] Error at 'return': Can't return from top-level code.
